File: 0215-kth-largest-element-in-an-array/0215-kth-largest-element-in-an-array.java
class Solution {

    private PriorityQueue<Integer> pq;

    public int findKthLargest(int[] nums, int k) {

        if (nums == null || nums.length == 0) return 0;

        int n = nums.length;

        pq = new PriorityQueue<>(n, new Comparator<Integer>() {
            public int compare(Integer a, Integer b) {
                return Integer.compare(b, a);
            }
        });

        IntStream.range(0, n).forEach(i -> {
            pq.offer(nums[i]);
        });

        IntStream.range(0, k - 1).forEach(i -> {
            pq.remove();
        });

        return pq.peek();
    }
}

File: 0215-kth-largest-element-in-an-array/0215-kth-largest-element-in-an-array.py
class Solution(object):
    def findKthLargest(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
#         nums = nums.sort()
#         temp = int(len(nums))
#         print(temp)
        
        return sorted(nums)[-k]
#         temp = nums[0]
# #         for j in range(1 , k):
# #             for i in range(1, len(nums)):
# #                 if nums[i] > temp: temp = nums[i]
            
# #             nums.remove(temp)
            
#         return max(nums)